pkg/skaffold/util/port.go

```diff
@@ -173,3 +173,27 @@ func IsPortFree(address string, p int) bool {
 	}
 	return true
 }
+
+// AllocatePort looks for a port close to desiredPort, using the provided implementation of
+// isPortAvailable to determine what ports can be used.
+// We deal with wrapping and avoid allocating ports < 1024
+func AllocatePort(isPortAvailable func(int32) bool, desiredPort int32) int32 {
+	var maxPort int32 = 65535 // ports are normally [1-65535]
+	if desiredPort < 1024 || desiredPort > maxPort {
+		log.Entry(context.TODO()).Debugf("skipping reserved port %d", desiredPort)
```

Comment:
We should pass a context into this function and do isCanceled() checks

Reply:
will require changes to several function signatures in the kubernetes debugging package. i'll add a TODO